upgrade-accounts.py

```diff
@@ -36,12 +36,13 @@ def main():
     accounts = []
     account_query = chain.query_map('System', 'Account', page_size=1000)
 
+    NEW_LOGIC_FLAG = 0x80000000_00000000_00000000_00000000
+
     for (i, (id, info)) in enumerate(account_query):
         account = info['data']
-        frozen = account['frozen']
-        reserved = account['reserved']
+        flags = account['flags'].decode()
 
-        if reserved > 0 or frozen > 0:
+        if flags & NEW_LOGIC_FLAG == 0:
```

Comment:
You still want to check for `reserved` or `frozen` existence. Else this check includes already migrated accounts.

Reply:
I think just checking for the flag should be enough. Migrating an account sets that flag [here](https://github.com/paritytech/polkadot-sdk/blob/0956357b602af1bb10f9bd6fb3998b856bd07592/substrate/frame/balances/src/lib.rs#L759-L762).

Reply:
Hey @wischli, I completely forgot about this PR.

Looking at it, I'm not sure how can already migrated accounts be included again?
The check is a duplicate from the `pallet-balances` code:
```Rust
const IS_NEW_LOGIC: u128 = 0x80000000_00000000_00000000_00000000u128;

pub fn is_new_logic(&self) -> bool {
	(self.0 & IS_NEW_LOGIC) == IS_NEW_LOGIC
}
```

Reply:
If you are talking about Centrifuge: It does not have #12951 deployed as it looks like. So the script is only needed once that is deployed.

Reply:
> If you are talking about Centrifuge: It does not have #12951 deployed as it looks like. So the script is only needed once that is deployed.

I ran this script against our testnet and an upgraded Chopstick fork of Centrifuge.

Reply:
> Hey @wischli, I completely forgot about this PR.
> 
> Looking at it, I'm not sure how can **already migrated accounts be included again**? The check is a duplicate from the `pallet-balances` code: [...]
> ```

Apologies for my poor phrasing. I was referring to the `master` branch script which includes migrated account due to the missing `flags` check.

After re-testing your script, it does indeed seem to  work. So another apologies here, I must have had a slight difference which introduced misbehaviour. Will close #2 then.

Reply:
No worries!